### PR TITLE
release build google maps

### LINKED ISSUE #4215: Map is not displaying
### Describe the issue:

- [ X ] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:
Since few development release, the google map is not displaying.

### Operating system:
MacOS 14.5 on MacBook Air M2

### Subsurface version:
6.0.5190-CICD-Release from subsurface daily 
Problem start with the 6.0.5185-CICD-Release, perhaps before.

### Steps to reproduce:
Launch the software, map is not displaying

### Current behavior:
Google map view is blank

### Expected behavior:
Google map view on the dive site

### Additional information:
Snapshot

### Mentions:
Capture d’écran 2024-05-28 à 08 37 03

### LINKED ISSUE #4225: Subsurface goes into an infinite loop when drilling into a Garmin MK3i dive with multiple transmitters
### Describe the issue:

- [ x] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:

In Subsurface 6.0.5194 CICD Release, I was able to import in dives from a Garmin MK3i with multiple T2 transmitters (3). However, after importing in, subsurface goes into an infinite loop when I import and drill into dives with multiple transmitters. Subsurface works as expected when I only import Garmin MK3i dives with a single transmitter. 

### Operating system:

macOS Sonoma 14.4.1

MacBook Air M2

### Subsurface version:

6.0.5194-CICD-Release

older versions cannot import Garmin MK3i dives

Official

### Steps to reproduce:

1) import Garmin MK3i FIT files 
2) go into the dives 

### Current behavior:

Subsurface goes into infinite loop when drilling into dives with multiple transmitters. Subsurface works as expected when I only import MK3i dives with a single transmitter.

### Expected behavior:

No infinite loop

### Additional information:

Yes, I have both Garmin MK3i dives with a single transmitter and others with multiple transmitters. I will upload them in a follow up. I am currently unable to upload files in this ticket. 

### Mentions:

@mikeller

### PR DESCRIPTION
**Pull request by notrege**
After the Mac QT upgrade to 5.15.13 google maps stopped working because a debug plugin was built and not deployed.   This changes forces a release build.   It may or may not be the best alternative, but if nothing else it's a starting point for discussion with people who know more about qmake than I do.

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
This changes forces a release build for the google maps plugin.   There may be a better way to do this (possibly using the scripts DEBUGRELEASE variable?), but I'm not certain of that.

### Related issues:

Fixes #4215.
Fixes #4225.

### CODE REVIEW

**Comment by Julaye** [comment]
This fix Works for me (I was the reporter of https://github.com/subsurface/subsurface/issues/4215) on artifacts [Subsurface-MacOS-6.0.5194-1-pull-request-googlemaps](https://github.com/subsurface/subsurface/suites/24388310471/artifacts/1556120159)

File: scripts/build.sh [hunk 1]
```diff
@@ -579,7 +579,7 @@ if [ "$QUICK" != "1" ] && [ "$BUILD_DESKTOP$BUILD_MOBILE" != "" ] && ( [[ $QT_VE
 		# regardless of whether we do a fat build or not, let's do the 'native' build here
 		$QMAKE "INCLUDEPATH=$INSTALL_ROOT/../qtlocation/build/include/QtLocation/6.3.0" QMAKE_APPLE_DEVICE_ARCHS="$(uname -m)" ../googlemaps.pro
 	else
-		$QMAKE "INCLUDEPATH=$INSTALL_ROOT/include" ../googlemaps.pro
+		$QMAKE "INCLUDEPATH=$INSTALL_ROOT/include" "CONFIG+=release" ../googlemaps.pro
```
Comment (mikeller) [inline scripts/build.sh:582]: I am wondering if we should be adding the same change on https://github.com/subsurface/subsurface/pull/4224/files#diff-3d64996115d0ac3b74e36f57ce6b00d6b4982614c41b1ed39eab8ef43b2e9a2bR580 - after all this is dependent on the Google maps plugin version, and we are using the same version for both paths.

**Review by mikeller — APPROVED** [review]

**Comment by mikeller** [comment]
Looking good, the map is showing when I run the pull request artifact on my Mac. Good work @notrege!

File: scripts/build.sh (hunk 1 above)
Comment (notrege) [reply scripts/build.sh:582]: No idea.  I don't have a QT 6 build environment so I have not tested to see if maps works or not.

Comment (mikeller) [reply scripts/build.sh:582]: I would add it - being explicit here will hopefully stop others from hitting the same (non-failing) bug when they try to build with Qt6.

**Pull request merged**